sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3469,9 +3469,14 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
       case _ =>
         // Note: table schema includes both the table columns list and the partition columns
         // with data type.
+        val tableProperties = TableProperties(properties, provider, options, location, comment,
+          serdeInfo, external)
         val schema = StructType(columns ++ partCols)
-        CreateTableStatement(table, schema, partitioning, bucketSpec, properties, provider,
-          options, location, comment, serdeInfo, external = external, ifNotExists = ifNotExists)
+        CreateV2Table(
```

Comment:
The naming is a bit hard here. The best name should be `CreateTable`, but it's already taken. `CreateV2Table` is not proper here as this can also create v1 tables.

Given that we can will migrate all the commands eventually, the existing `CreateTable` will be removed at the end. How about we still use the name `CreateTable` here? i.e. add a new `CreateTable` logical plan in `org.apache.spark.sql.catalyst.plans.logical`, which does not conflict with the existing `CreateTable`